### PR TITLE
D8CORE-5777 Correct display if missing fields

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- Corrected template logic to account for possibly missing fields in the header

# Review By (Date)
- Thursday

# Criticality
- 2

# Review Tasks

## Setup tasks and/or behavior to test

1. Checkout this branch.
2. Create a course using the content type.  Omit the academic year and/or the course code.
3. Verify that card and list displays for the course are correct, whether or not those fields are present.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: templates/components/course-list/course-item-list.twig [hunk 1]
```diff
@@ -48,11 +48,14 @@
         {%- endif -%}
   </div>
 
+  {%- if course_list_code|render_clean is not empty or course_list_academic_year|render_clean is not empty -%}
   <header class="su-course-list__header">
-    {%- if course_list_code is not empty -%}<strong>{{ course_list_code|render|striptags|replace({'\n': '', ' ': ''}) }}</strong>{%-  endif -%}&nbsp;|{%- if course_list_academic_year is not empty -%}{{ course_list_academic_year }}{%-  endif -%}
-
+    {%- if course_list_code is not empty -%}<strong>{{ course_list_code|render|striptags|replace({'\n': '', ' ': ''})}}</strong>{%-  endif -%}
```
Comment (jenbreese) [inline templates/components/course-list/course-item-list.twig:53]: If the Academic Year field is filled in, the the `strong` tag is displaying as empty where the Course Code would be.

Comment (imonroe) [reply templates/components/course-list/course-item-list.twig:53]: Ah, the empty check needed the `render_clean` to really determine if it was empty.

Comment (imonroe) [reply templates/components/course-list/course-item-list.twig:53]: yeah, that did the trick.  
image

**Review by jenbreese — APPROVED** [review]
works as wanted.

**Pull request merged**